### PR TITLE
[WIP] Optimize write step by skipping convert to 3 steps

### PR DESCRIPTION
**Pull request by ajantha-bhat**
### Why is this PR needed?
 
 
 ### What changes were proposed in this PR?

    
 ### Does this PR introduce any user interface change?
 - No
 - Yes. (please explain the change and update document)

 ### Is any new testcase added?
 - No
 - Yes

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.4/749/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2457/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.4/766/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2474/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.4/774/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2481/

File: processing/src/main/java/org/apache/carbondata/processing/loading/sort/SortStepRowHandler.java [hunk 1]
```diff
@@ -192,6 +196,16 @@ public SortStepRowHandler(SortParameters sortParameters) {
     return out;
   }
 
+  public Object[] convertToFlatRow(IntermediateSortTempRow sortTempRow) {
+    List<Object> out = new ArrayList<>();
```
Comment (VenuReddy2103) [inline processing/src/main/java/org/apache/carbondata/processing/loading/sort/SortStepRowHandler.java]: Suggest to create with initialCapacity as the number of elements are at this point.

Comment (ajantha-bhat) [reply processing/src/main/java/org/apache/carbondata/processing/loading/sort/SortStepRowHandler.java]: @VenuReddy2103 : This is WIP PR, I am working on it now. As effort is more.

**Comment by ajantha-bhat** [comment]
This is WIP PR, **I am NOT working on it now.** as it take so much time to finish and stabilize it

**Pull request closed without merging**